### PR TITLE
Add Debian OS family support

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -59,6 +63,13 @@
         "8",
         "9"
       ]
+    },
+    {
+      "operatingsystem": "Ubuntu",
+      "operatingsystemrelease": [
+        "20.04",
+        "22.04"
```
Comment (bastelfreak) [inline metadata.json:71]: bolt is still missing for Debian 12 / Ubuntu 24.04 :(
* https://github.com/puppetlabs/bolt/issues/3280
* https://github.com/puppetlabs/bolt/issues/3281

File: manifests/init.pp [hunk 2]
```diff
@@ -32,15 +32,15 @@
 #
 class bolt (
   String[1] $version = 'installed',
-  Stdlib::HTTPSUrl $base_url = 'https://yum.puppet.com/',
-  String[1] $release_package = "puppet-tools-release-el-${facts['os']['release']['major']}.noarch.rpm",
-  String[1] $gpgkey = 'RPM-GPG-KEY-puppet-20250406',
-  Boolean $use_release_package = true,
+  Stdlib::HTTPSUrl $base_url = $facts['os']['family'] ? { 'Debian' => 'https://apt.puppet.com/', 'RedHat' => 'https://yum.puppet.com/', },
```
Comment (rwaffen) [inline manifests/init.pp:35]: i do not like this. not at all. but have no better idea. for my understanding this should be elaborated int the class body

Comment (bastelfreak) [reply manifests/init.pp:35]: If I do it in the body, the user cannot overwrite it.

Comment (rwaffen) [reply manifests/init.pp:35]: yeah, as i say, i just dont like the style, feel free to proceed 😅

**Review by rwaffen — APPROVED** [review]

**Pull request merged**